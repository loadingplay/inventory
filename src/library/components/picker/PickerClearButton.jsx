import React from 'react';
import { PickerStore } from '../../services';

export class PickerClearButton extends React.Component {
  static defaultProps = {
    onPickerClear: () => {}
  }

  constructor(props) {
    super(props);
  }

  handleClear = () => {
    PickerStore.clear(this.props.selected_cellar_id);
    this.props.onPickerClear();
  }

  render = () => {
    return (
      <button type="button" onClick={this.handleClear} className="btn btn-danger btn-block float-right" >
        limpiar picker
      </button>
    );
  }
}
