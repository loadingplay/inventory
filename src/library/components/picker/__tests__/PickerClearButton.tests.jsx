import React from 'react';
import { shallow } from "enzyme";

import { PickerClearButton } from "../PickerClearButton";

describe("PickerClearButton", () => {
  test("it should render properly", () => {
    const wrapper = shallow(<PickerClearButton></PickerClearButton>);
    expect(wrapper).toMatchSnapshot("detault");
  });
});
