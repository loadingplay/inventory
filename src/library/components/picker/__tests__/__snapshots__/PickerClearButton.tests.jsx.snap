// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerClearButton it should render properly: detault 1`] = `
<button
  className="btn btn-danger btn-block float-right"
  onClick={[Function]}
  type="button"
>
  limpiar picker
</button>
`;
